tests/acceptance/pageObjects/sharedWithMePage.js

```diff
@@ -16,38 +16,31 @@ module.exports = {
       )
     },
     /**
+     * gets displayed share status of file-name (shared by user of given username)
+     *
      * @param {string} filename
-     * @param {string} status - It takes one of the following : declined, pending or '' for accepted
      * @param {string} user
-     * Checks if the file-row of the desired file-name with the username consists of the desired status of accepted,
-     * declined or pending
-     */
-    assertDesiredStatusIsPresent: function (filename, status, user) {
-      let requiredXpath = this.api.page.FilesPageElement.filesList().getFileRowSelectorByFileName(filename) +
-                          util.format(this.elements.assertStatusFileRow.selector, status)
-      requiredXpath = user === undefined ? requiredXpath : requiredXpath +
-                      util.format(this.elements.getSharedFromUserName.selector, user)
-      return this.waitForElementVisible({
-        locateStrategy: this.elements.assertStatusFileRow.locateStrategy,
-        selector: requiredXpath
-      })
-    },
-    /**
-     * @param {string} filename
-     * @param {string} sharer
-     * @param {string} status - It takes one of the following : declined, pending or '' for accepted
-     * Checks if the file-row of the desired file-name with the username doesn't consist of the desired status of accepted,
-     * declined or pending
+     *
+     * @return {Promise<string>}
      */
-    assertDesiredStatusIsAbsent: function (filename, sharer, status) {
-      let requiredXpath = this.api.page.FilesPageElement.filesList().getFileRowSelectorByFileName(filename) +
-        util.format(this.elements.assertStatusFileRow.selector, status)
-      requiredXpath = sharer === undefined ? requiredXpath : requiredXpath +
-        util.format(this.elements.getSharedFromUserName.selector, sharer)
-      return this.waitForElementNotPresent({
-        locateStrategy: this.elements.assertStatusFileRow.locateStrategy,
-        selector: requiredXpath
-      })
+    getShareStatusOnFileRow: async function (filename, user) {
+      let status
+      let requiredXpath = this.api.page.FilesPageElement.filesList().getFileRowSelectorByFileName(filename)
+      requiredXpath = user === undefined ? requiredXpath
+        : requiredXpath + util.format(this.elements.getSharedFromUserName.selector, user) + '/../..'
+      requiredXpath += this.elements.shareStatusOnFileRow.selector
+      await this
+        .useXpath()
+        .waitForAnimationToFinish()
+        .api.getText(requiredXpath,
+          (result) => {
+            if (result.status === 0) {
+              status = result.value === '' ? 'Accepted' : result.value
+            }
```

Comment:
Probably needs error handling here.

Reply:
@dpakach Yes, I've added a line to throw new error in new commit, please review!